app/src/main/java/org/mozilla/fenix/compose/button/RadioButton.kt

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.button
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.material.RadioButtonColors
+import androidx.compose.material.RadioButtonDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Radio Button.
+ *
+ * @param selected [Boolean] showing if the radio button is selected or not.
+ * @param modifier [Modifier] to be applied to the radio button.
+ * @param enabled [Boolean] that controls if radio button is selectable.
+ * @param onClick Callback to be invoked when radio button is clicked.
+ */
+@Composable
+fun RadioButton(
+    selected: Boolean,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    colors: RadioButtonColors = RadioButtonDefaults.colors(
+        selectedColor = FirefoxTheme.colors.iconActionSecondary,
+        unselectedColor = FirefoxTheme.colors.iconActionSecondary,
+        disabledColor = FirefoxTheme.colors.iconDisabled,
+    ),
```

Comment:
@gabrielluong, what are your thoughts with this component? Should we leave it as _just_ the radio button for now, or should we take this opportunity to add optional primary/secondary strings so this could be consumed in a settings/preference screen?

The figma has this in two use cases:
1. With just one piece of text
2. Just the button

Reply:
Let's leave it as follow up. Reviewing less amount of code at a time is always preferable from my pov. I would rather just focus on getting what we have added here right instead of adding more and having the new requirements cause further review churns.